### PR TITLE
L46: Python Compression API

### PR DESCRIPTION
**Pull request by gnossen**

### CODE REVIEW

**Comment by carl-mastrangelo** [comment]
fyi you should probably change the PR title too.   When other people are looking for the next proposal number, they might look at the repo file names and open PRs.

File: L46-python-compression-api.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+    @abc.abstractmethod
+    def __call__(self,
+                 request_iterator,
+                 timeout=None,
+                 metadata=None,
+                 credentials=None,
+                 wait_for_ready=None,
+                 compression=None):
+        ...
+```
```
Comment (lidizheng) [inline L46-python-compression-api.md]: The interceptor API interface have to change as well. Don't forget the [ClientCallDetails](https://grpc.io/grpc/python/grpc.html#grpc.ClientCallDetails).

**Comment by gnossen** [comment]
@lidizheng PTAL.

File: L46-python-compression-api.md [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+Other implementations offer richer APIs that allow a user to define their own
+compression algorithms. However, given the structure of gRPC core, it would not
+be simple to implement this for gRPC Python. We are essentially limited to the
+options offered by core. The proposal detailed here simply surfaces those
+interfaces for Python users.
+
+
+## Implementation
+
+I (@gnossen) will implement these changes and merge them after the proposal has
```
Comment (lidizheng) [inline L46-python-compression-api.md]: Let's link the implementation PR

**Pull request merged**